### PR TITLE
feat: Add gitlab preset

### LINKED ISSUE #2654: Support GitLab presets
**What would you like Renovate to be able to do?**
https://renovatebot.com/docs/config-presets/#how-to-publish-preset-configs describes how to share common configuration across multiple repositories. By doing `github>owner/name` you can use configs that are hosted via GitHub without publishing to packagist. I'd like to do the same for GitLab.

**Describe the solution you'd like**
Basically a clone of GitHub's feature but for GitLab repositories: `gitlab>owner/name`.

**Describe alternatives you've considered**
I thought about publishing to NPM but I believe my configs could contain trade secrets.

**Additional context**
I would also like to mention that GitLab does also support `vendor/group/repo` and `vendor/group/subgroup/repo` for their repository URLs, this differs from GitHub.

### PR DESCRIPTION
**Pull request by rtaum**
Adds gitlab hosted preset config

Closes #2654

### CODE REVIEW

File: lib/datasource/gitlab.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+  if (presetName !== 'default') {
+    throw new Error(
+      { pkgName, presetName },
+      'Sub-preset names are not supported with Gitlab datasource'
+    );
+  }
+  let res;
+  try {
+    const urlEncodedPkgName = encodeURIComponent(pkgName);
+    const url = `https://gitlab.com/api/v4/projects/${urlEncodedPkgName}/repository/files/renovate.json?ref=master`;
```
Comment (rarkins) [inline lib/datasource/gitlab.js]: As per @rtaum, he thinks GitLab makes us pick a branch. Ideally we could find a way to just tell GitLab "the default branch", in case it is not master. So let's see if we can figure out a way. Maybe it even takes one extra request to do?

**Comment by rarkins** [comment]
@Jamesking56 can you check out this branch and see if it works as you'd expect?

File: lib/datasource/gitlab.js (hunk 1 above)
Comment (Jamesking56) [reply lib/datasource/gitlab.js]: You can do this by doing an API call to `v4/projects/:id/repository/branches`:

https://docs.gitlab.com/ee/api/branches.html

**Comment by rarkins** [comment]
In that case we can query the branches api endpoint first and cache the default branch for perhaps 24 hours, then query and cache the actual preset for something like 15 minutes.

**Comment by rarkins** [comment]
I might merge this as-is though. Non-master repos should be an edge case and fine to implement in a future PR.

**Comment by Jamesking56** [comment]
Reading through it I think it'll be alright to merge to begin with and then I can test it on my renovate setup :+1:
